.xe/features/engineering-context/spec.md

```diff
@@ -0,0 +1,132 @@
+
+## Goals
+
+- Provide token-efficient templates for essential architecture/engineering-owned context (architecture, engineering principles, development process)
+- Enable AI agents to make technical implementation decisions aligned with architecture patterns and engineering standards
+- Minimize token usage while maintaining clarity and decision-making utility
+
+Explicit non-goals:
+
+- This feature does NOT include standards files (`.xe/standards/` - separate concern)
+- This feature does NOT include playbook templates (separate feature)
+- Development process template is included but lives in `.xe/process/` directory (engineering-owned context)
+
+## Scenario
+
+- As an **Architect**, I need a concise template for documenting technology stack and architecture patterns so that AI can make aligned technical decisions
+  - Outcome: Token-efficient architecture documentation guiding all technical implementation
+
+- As an **Engineer**, I need a template for engineering principles that AI can consume without excessive token overhead so that implementation follows consistent standards
+  - Outcome: Clear engineering principles without verbose instructions or duplicate content
+
+- As an **AI Agent**, I need to read architecture and engineering context during feature implementation so that I can align with technical decisions and quality standards
+  - Outcome: AI-generated code aligns with architecture patterns and engineering principles
+
+## Success Criteria
+
+- Architecture.md template provides comprehensive technical architecture guidance while maintaining token efficiency
+- Engineering.md template provides actionable engineering principles and technical standards
+- Templates follow [template standard](.xe/standards/catalyst.md)
+- Templates enable AI agents to make technical decisions without reading unnecessary context
+
+## Design principles
+
+See [.xe/product.md](.xe/product.md) for product-wide design principles.
+
+No feature-specific design principles are needed for this template feature.
+
+## Requirements
+
+### Functional Requirements
+
+- **FR-1**: Architecture.md template MUST exist
+  - **FR-1.1**: Template MUST follow template standard defined in `.xe/standards/catalyst.md`
+  - **FR-1.2**: Template MUST include Overview section with pointers to related context files
+  - **FR-1.3**: Template MUST include Technology Stack section with Runtime and Development subsections:
+    - **FR-1.3.1**: Runtime Technologies subsection with deployment footprint aspects:
+      - Runtime Environment
+      - Runtime Dependencies
+      - Data Storage
+      - AI Integration (Runtime)
+      - Deployment Method
+      - Security
+      - Monitoring/Logging
+    - **FR-1.3.2**: Development Technologies subsection with build-time tool aspects:
+      - Dev Dependencies
+      - Automation
+      - AI Integration (Dev)
+      - Testing Framework
```

Comment:
Do we have everything that would normally be accounted for in an dev dependencies? Here are a standard set of cloud resource categories, but I don't know that we need all of them:
- AI Code Gen
- Backup and Disaster Recovery
- Compliance
- Continuous Integration and Deployment
- Cost Management
- Data Governance
- Developer Platform
- Development Environment
- Endpoint Management
- Observability
- Testing Framework
- Source Code Management

As I said about runtime tech, I'm a fan of including everything to minimize risk of AI missing something and having AI just remove what isn't needed.